src/main/clojure/clara/rules/durability/fressian.clj

```diff
@@ -535,17 +539,34 @@
           indexed-facts))))
 
   (deserialize [_ mem-facts opts]
+    ;; Options can be provided via opts.
+    ;; These include:
+
+    ;; * :fact-type-fn - Documented in clara.rules/mk-session
+    ;; * :activation-group-sort-fn - Documented in clara.rules/mk-session
+    ;; * :activation-group-fn - Documented in clara.rules/mk-session
+    ;; * :rulebase-only? - If true only deserialize a rulebase, not a session.
+    ;; * :base-rulebase - Rulebase to use with a deserialized session.
+
     (with-open [^FressianReader rdr (fres/create-reader in-stream :handlers read-handler-lookup)]
       (let [{:keys [rulebase-only? base-rulebase]} opts
 
             record-holder (ArrayList.)
             ;; The rulebase should either be given from the base-session or found in
             ;; the restored session-state.
-            rulebase (or (and (not rulebase-only?) base-rulebase)
-                         (binding [d/*node-id->node-cache* (volatile! {})
-                                   d/*clj-record-holder* record-holder
-                                   d/*compile-expr-fn* (memoize (fn [id expr] (com/try-eval expr)))]
-                           (fres/read-object rdr)))]
+            maybe-base-rulebase (when (and (not rulebase-only?) base-rulebase)
+                                  base-rulebase)
+
+            rulebase (if maybe-base-rulebase
+                       maybe-base-rulebase
+                       (let [without-ops-rulebase (binding [d/*node-id->node-cache* (volatile! {})
+                                                            d/*clj-record-holder* record-holder
+                                                            d/*compile-expr-fn* (memoize (fn [id expr] (com/try-eval expr)))]
+                                                    (fres/read-object rdr))]
+                         (assoc without-ops-rulebase
```

Comment:
Yes, I meant pulling that part out.  The idea is to have things that are SerDe implementation independent in `clara.rules.durability`.  So it'd be a start.

I'd like to have the implementation of Fressian SerDe to do only things that are specific to Fressian as much as possible.

However, it's mostly just organizational.  It wouldn't be expected that there'd be a lot of different implementations of `d/ISessionSerializer` since it is highly tied to the impl details.  However, if we ended up wanting to add more than just Fressian support, it'd be nice to have the implementations as thin as possible.